pkg/events/event_store.go

```diff
@@ -66,7 +66,9 @@ func (es *EventStore) CollectEvents() []*si.EventRecord {
 	es.Lock()
 	defer es.Unlock()
 
-	messages := es.events[:es.idx]
+	messages := make([]*si.EventRecord, len(es.events[:es.idx]))
```

Comment:
Could you add test to validate the copy?

Reply:
+1, let's make sure that changes to the original are not reflected in the copy

Reply:
Done